tests/API/Management/ClientsTest.php

```diff
@@ -2,42 +2,197 @@
+     *
+     * @var string
+     */
+    protected $id_name = 'client_id';
+
+    /**
+     * Name of the created Client.
+     * Appended with a random string in self::__construct().
+     *
+     * @var string
+     */
+    protected $create_client_name = 'TEST-CREATE-CLIENT-';
+
+    /**
+     * Application type of the created Client.
+     *
+     * @var string
+     */
+    protected $create_app_type = Clients::APP_TYPE_REGULAR_WEB;
+
+    /**
+     * SSO setting of the created Client.
+     *
+     * @var bool
+     */
+    protected $create_sso = false;
+
+    /**
+     * Description of the created Client.
+     *
+     * @var string
+     */
+    protected $create_desc = '__Auth0_PHP_initial_app_description__';
 
-    protected function getApiClient() {
-        $env = $this->getEnv();
-        $token = $this->getToken($env, [
-            'clients' => [
-                'actions' => ['create', 'read', 'delete', 'update']
-            ]
-        ]);
+    /**
+     * Name of the updated Client.
+     * Appended with a random string in self::__construct().
+     *
+     * @var string
+     */
+    protected $update_client_name = 'TEST-UPDATE-CLIENT-';
 
-        $this->domain = $env['DOMAIN'];
+    /**
+     * Application type of the updated Client.
+     *
+     * @var string
+     */
+    protected $update_app_type = Clients::APP_TYPE_NON_INTERACTIVE;
 
-        $api = new Management($token, $env['DOMAIN']);
+    /**
+     * SSO setting of the updated Client.
+     *
+     * @var bool
+     */
+    protected $update_sso = true;
 
+    /**
+     * Description of the updated Client.
+     *
+     * @var string
+     */
+    protected $update_desc = '__Auth0_PHP_updated_app_description__';
+
+    /**
+     * ClientsTest constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->create_client_name .= rand();
+        $this->update_client_name .= rand();
+    }
+
+    /**
+     * Return the ID for the entity created.
+     *
+     * @param array $entity - Client created during test.
+     *
+     * @return string|integer
+     */
+    protected function getId($entity)
+    {
+        return $entity[$this->id_name];
+    }
+
+    /**
+     * Return the Clients API to test.
+     *
+     * @return Management\Clients
+     */
+    protected function getApiClient()
+    {
+        $token = $this->getToken($this->env, [ 'clients' => [ 'actions' => ['create', 'read', 'delete', 'update' ] ] ]);
+        $api = new Management($token, $this->domain);
         return $api->clients;
     }
 
-    protected function getCreateBody() {
-        $client_name = 'test-create-client' . rand();
-        echo "\n-- Using client name $client_name \n";
+    /**
+     * Get the Client create data to send with the test create call.
+     *
+     * @return array
+     */
+    protected function getCreateBody()
+    {
+        return [
+            'name' => $this->create_client_name,
```

Comment:
why not just put all this body data on a JSON file and read it or even a single parameter instead of having many of them defined above? it looks much cleaner IMO.